test/inference/advi_distributionsad.jl

```diff
@@ -0,0 +1,83 @@
+
+const PROGRESS = length(ARGS) > 0 && ARGS[1] == "--progress" ? true : false
+
+using Test
+
+@testset "advi" begin
+    @testset "locscale" begin
+        @testset "$(modelname) $(objname) $(realtype) $(adbackname)"  for
+            realtype ∈ [Float64], # Currently only tested against Float64
+            (modelname, modelconstr) ∈ Dict(
+                :NormalLogNormalMeanField => normallognormal_meanfield,
+            ),
+            (objname, objective) ∈ Dict(
+                :ADVIClosedFormEntropy  => ADVI(10),
+                :ADVIStickingTheLanding => ADVI(10, entropy = StickingTheLandingEntropy()),
+            ),
+            (adbackname, adbackend) ∈ Dict(
+                :ForwarDiff  => AutoForwardDiff(),
+                # :ReverseDiff => AutoReverseDiff(),
+                # :Zygote      => AutoZygote(), 
+                # :Enzyme      => AutoEnzyme(),
```

Comment:
Why are we not testing any of the other AD backends?

Reply:
`Enzyme` doesn't work yet with `Distributions`, `Zygote` has a bug with `mapreduce` that hasn't been resolved, `ReverseDiff` seems to have issues with `DistributionsAD`, but not sure about this one. I'll take a deeper look. But overall, the AD side of things is not in good shape :worried: